src/app/modules/common/services.js

```diff
@@ -535,6 +555,65 @@ angular.module('BloomLibraryApp.services', ['restangular'])
                 rBook.customPUT(putParams);
             };
 
+        this.relateBooksById = function() {
+            //Context of arguments is lost going into IIFE
+            var tArgs = arguments;
+
+            //Since, it might be possible for the user to click a second book
+            //faster than the server can process the request, I'm using an IIFE
+            (function() {
+                var args = tArgs;
+                var relatedBooks = [];
+
+                var baseRelatedBooks = restangular.withConfig(authService.config()).all('classes/relatedBooks');
+
+                var i = 0;
+
+                function handleOneId() {
+                    //Base case, out of arguments
+                    if(i >= args.length) {
+                        //Only create new relationship if two or more books are involved
+                        if(relatedBooks.length > 1) {
+                            //Create new entry with relationship
+                            baseRelatedBooks.post({"books": relatedBooks});
+                        }
+                    }
+                    else {
+                        //Add this book pointer to the queue
+                        relatedBooks.push({
+                            "__type": "Pointer",
+                            "className": "books",
+                            "objectId": args[i]
+                        });
+                        baseRelatedBooks.getList({
+                            'where': {
+                                'books': {
+                                    "__type": "Pointer",
+                                    "className": "books",
+                                    "objectId": args[i]
+                                }
+                            }
+                        }).then(function (results) {
+                            if (results.length > 0) {
+                                //For some reason, I was unable to use results[0].remove()
+                                restangular.withConfig(authService.config()).one('classes/relatedBooks', results[0].objectId).remove().then(function () {
```

Comment:
I think the effect of the line above is that this method will remove any existing relationships involving any of the books in the new one. That ought to be stated in its comment, as it is a non-obvious side effect not indicated in the method name.